src/main/scala/mesosphere/marathon/core/launcher/impl/OfferProcessorImpl.scala

```diff
@@ -121,10 +121,15 @@ private[launcher] class OfferProcessorImpl(
   private[this] def saveTasks(ops: Seq[TaskOpWithSource], savingDeadline: Timestamp): Future[Seq[TaskOpWithSource]] = {
     def saveTask(taskOpWithSource: TaskOpWithSource): Future[Option[TaskOpWithSource]] = {
       val taskId = taskOpWithSource.taskId
-      log.info("Save task [{}]", taskOpWithSource.taskId)
-      taskCreationHandler
-        .created(taskOpWithSource.op.newTask)
-        .map(_ => Some(taskOpWithSource))
+      val persistedOp = taskOpWithSource.op.maybeNewTask match {
+        case Some(newTask) =>
```

Comment:
I would prefer to extend the logging statements if that's ok.

What information is missing for you?

What do you think of

```scala
      val persistedOp = taskOpWithSource.op.maybeNewTask match {
        case Some(newTask) =>
          log.info(
            "Save {} after applying the effects of {}",
            taskOpWithSource.taskId, taskOpWithSource.op.getClass.getSimpleName
          )
          taskCreationHandler.created(newTask)
        case None =>
          log.info("Remove {} because of {}", taskOpWithSource.taskId, taskOpWithSource.op.getClass.getSimpleName)
          taskCreationHandler.terminated(taskId)
      }
```